controllers/Handle.php

```diff
@@ -137,51 +195,38 @@ public function callback(string $provider): RedirectResponse
             'user_type' => get_class($user),
             'user_id' => $user->getKey(),
             'provided_id' => $ssoUser->getId(),
-            'provided_email' => $ssoUser->getEmail(),
+            'provided_email' => $email,
             'ip' => Request::ip(),
             'metadata' => [
                 'remember' => $remember,
             ],
         ]);
 
-        // @TODO: Handle this via an event listener on the backend.auth.login event
-        // and ensure that the event is fired by the AuthManager
-        $runMigrationsOnLogin = (bool) Config::get('cms.runMigrationsOnLogin', Config::get('app.debug', false));
-        if ($runMigrationsOnLogin) {
-            try {
-                // Load version updates
-                UpdateManager::instance()->update();
-            } catch (Exception $e) {
-                Flash::error($e->getMessage());
-            }
-        }
-
-        // @TODO: Also handle via event listener
-        // Log the sign in event
-        AccessLog::add($user);
-
         // Redirect to the intended page after successful sign in
         return Backend::redirectIntended('backend');
     }
 
     /**
      * Redirects the user to the authentication page of the given provider.
+     * Redirects back to signin form on errors with Flash message.
+     * @throws SystemException if the SSO provider is not enabled or is misconfigured.
      */
     public function redirect(string $provider): RedirectResponse
     {
         if (!in_array($provider, $this->enabledProviders)) {
-            return $this->redirectToSigninPage("The {$provider} SSO provider is not enabled");
+            throw new SystemException("The $provider SSO provider is not enabled.");
```

Comment:
So that the error gets logged. How does the user even end up on this code path if the provider is disabled?

Reply:
> So that the error gets logged.

Then we should add a log statement before redirecting back to the signin page.

> How does the user even end up on this code path if the provider is disabled?

They shouldn't, unless something's not right